### PR TITLE
Graph tests

### PR DESCRIPTION
**Pull request by microgravitas**
This addresses issue #43: the TFGraph class and the code to compute dominating sets are now tested with random graph instances and some small hand-crafted cases. The dominating set code contains more comments, however, without knowledge of the underlying algorithms/data-structures even these will probably not help.

### CODE REVIEW

File: spacegraphcats/rdomset.py [hunk 1]
```diff
@@ -184,19 +192,23 @@ def ldo(g, weight=None):
         seen.add(v)
     return res
 
-""" Dvorak's algorithm with dtf-augs instead of weakly linked cols """
-
 def calc_distance(augg, X):
```
Comment (domoritz) [inline spacegraphcats/rdomset.py]: This function is not tested, right?

File: spacegraphcats/rdomset.py [hunk 2]
```diff
@@ -239,6 +256,9 @@ def calc_dominator_sizes(g, domset, d):
     return res
 
 def calc_dominated(augg, domset, d):
```
Comment (domoritz) [inline spacegraphcats/rdomset.py]: I don't think this is tested either.

File: spacegraphcats/rdomset.py [hunk 3]
```diff
@@ -344,21 +367,29 @@ def better_dvorak_reidl(augg,d):
 
     return domset
 
-def test_scattered(g, sc, d):
+#
+# Debugging/Test functions
+#
+def verify_scattered(g, sc, d):
```
Comment (domoritz) [inline spacegraphcats/rdomset.py]: Even though this is a debugging function, why not call it from a test?

File: spacegraphcats/rdomset.py [hunk 4]
```diff
@@ -369,7 +400,7 @@ def test_dominators(g, domset, dominators, d):
             return False
     return True
 
-def test_dominators_strict(g, domset, dominators, d):
+def verify_dominators_strict(g, domset, dominators, d):
```
Comment (domoritz) [inline spacegraphcats/rdomset.py]: same as above

**Comment by domoritz** [comment]
In general, we use (4) spaces instead of tabs.

**Comment by microgravitas** [comment]
Sorry, I'll check my editor settings.

File: spacegraphcats/rdomset.py (hunk 1 above)
Comment (microgravitas) [inline spacegraphcats/rdomset.py]: Addressed in the latest commit.

File: spacegraphcats/rdomset.py (hunk 2 above)
Comment (microgravitas) [inline spacegraphcats/rdomset.py]: Addressed in the latest commit.

File: spacegraphcats/rdomset.py (hunk 3 above)
Comment (microgravitas) [inline spacegraphcats/rdomset.py]: Addressed in the latest commit.

File: spacegraphcats/rdomset.py (hunk 4 above)
Comment (microgravitas) [inline spacegraphcats/rdomset.py]: Addressed in the latest commit.

**Pull request merged**